### ISSUE TITLE
Wrong redirect on Samsung-Browser

### ISSUE THREAD

**Issue by jenswittmann** [Jako/LangRouter#21]
Hello @Jako ,
i detect a wrong redirect on Samsung-Browser. The Browser sends the correct Languages with `$_SERVER['HTTP_ACCEPT_LANGUAGE']`, but the result is flip before use. I send a pull request for my change, now it works. Can you tell me why the array is fliped?

**Comment by Jako** [comment]
The following structure is returned by clientLangDetect:

```
Array
(
    [de-DE] => 1
    [de] => 0.9
    [en] => 0.8
)
``` 

In contextKeyDetect, the culture key of a context has to be found in that array. This is done with preg_grep, which works in array values. Because of this the array is flipped.

There could occur something not expected in the current code, if two contexts contain the same culture_key. Then the contextmap will be filled with the last context with this culture_key. But which context should be selected in that case? Now it is the last context. It could be changed to the first context. But will that be better?

So please look whats returned by clientLangDetect. Is the structure the same? If not, please check the regular expression in the clientLangDetect method.

**Comment by jenswittmann** [comment]
Hello Jako,
thanks for your explanation! I test it again. The problem is, that the value for the q value is not sended by the the samsung browser. So it will be set by LangRouter as value `1`.
```
Array (  
    [de-DE] => 1  
    [de] => 1 
    [en-US] => 1
    [en] => 1  
)
```

In this case, it all get into one by array_flip() and set to the last entry, the `en` value. I thinks its a problem from the samsung browser, but it would be nice to handle that. In the clientLangDetect() on line 206 is the q default value set. Maybe we can set a better value like this?

```
// set default any without q factor
$i = 1;
foreach ($langs as $lang => $val) {
    if ($val === '') {
        $langs[$lang] = $i;
        $i = $i - 0.1;
    }
}
```

**Comment by Jako** [comment]
I hope this is a bit more suitable (assuming the list is ordered by the q factor and in between are not detected q factors - I have seen only q factors with 1 decimal at the moment):
```
// set default to 1 (or the last value decremented by 0.01) for any language without q factor
$q = 1;
foreach ($langs as $lang => $val) {
    if ($val === '') {
        $langs[$lang] = $q;
        $q -= 0.01;
    } else {
        $q = $val - 0.01;
    }
}
```

**Comment by jenswittmann** [comment]
Looks good, can be added i think. Thanks for the great help!

**Comment by Jako** [comment]
Did it work for you?

**Comment by jenswittmann** [comment]
> Did it work for you?

Yes, for sure.

**Comment by Jako** [comment]
Fixed with 1.3.2
